server/etcdmain/grpc_proxy.go

```diff
@@ -218,7 +218,7 @@ func startGRPCProxy(cmd *cobra.Command, args []string) {
 	httpClient := mustNewHTTPClient(lg)
 
 	srvhttp, httpl := mustHTTPListener(lg, m, tlsinfo, client, proxyClient)
-	errc := make(chan error)
+	errc := make(chan error, 3)
```

Comment:
Could you, please,  use errgroup instead: 

https://pkg.go.dev/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup

It seems to be canonical application of this infrastructure.

Reply:
@lzhfromustc you may look at

https://github.com/etcd-io/etcd/blob/7f97dfd45ac82518716a59c246e2c2a95c9825c7/tests/integration/v3_leadership_test.go#L213-L222

for an inspiration